temporalio/worker/workflow_sandbox/_in_sandbox.py

```diff
@@ -39,7 +39,37 @@ def __init__(
         # class. We can't use the definition that was given to us because it has
         # type hints and references to outside-of-sandbox types.
         new_defn = temporalio.workflow._Definition.must_from_class(workflow_class)
-        new_instance_details = dataclasses.replace(instance_details, defn=new_defn)
+
+        # Also, we have to re-import the worker-level exception types, because
+        # some exceptions are not passthrough and therefore our issubclass fails
+        # because it'll be comparing out-of-sandbox types with in-sandbox types
```

Comment:
eww, that kinda sucks. Not that there's anything to be done about it really I guess, but, eww.

Reply:
I couldn't think of a better way and I'm sure TS will have this problem too. Basically we want to support subtype checking for this, but in some cases Python sandbox reloads a user type inside the sandbox and it'll fail to match what was set outside the sandbox.

Another option here would be to not allow this to be set on the worker but instead make a new interceptor method that could be overridden to determine whether an exception was task fail or workflow fail. But I figured reloading was the best from a user POV.

Reply:
It is a bit fragile for advanced cases, but this is what that "prepare workflow" concept is for. It creates a workflow in the sandbox on worker start just as if it got work for that workflow in an attempt to fail quickly for anything not sandbox compatible.